tests/models/test_amp.py

```diff
@@ -190,8 +191,15 @@ def test_amp_without_apex(tmpdir):
 @pytest.mark.skipif(not _APEX_AVAILABLE, reason="test requires apex")
 def test_amp_with_apex(tmpdir):
     """Check calling apex scaling in training."""
+    class CustomModel(EvalModelTemplate):
```

Comment:
@rohitgr7 I had to keep this EvalModelTemplate
To the best of my abilities, I am  just not able to figure out why boring model with multi optimizers does not but eval template does (in the case of apex). 

I propose to leave it for now. In the meantime, I have opened an issue here: #5642

Reply:
interesting... on master it's passing with BoringModel. The fix Thomas is working is to ensure that requires_grad for layers remain same.

Reply:
ok, maybe the PR with master commits will be merged soon into 1.2 so we will check again when that happens :)